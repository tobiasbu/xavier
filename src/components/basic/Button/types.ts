
/**
 * Astro colors
 */
type ButtonColor = 'uranus' | 'venus' | 'moon' | 'neptune' | 'mars' | 'earth';

/**
 * Describe button properties
 */
export interface ButtonProps {
  /**
   * Button color
   */
  color?: ButtonColor;
  /**
   * Button text
   */
  children?: string;
  /**
   * Custom class name for button container.
   */
  className?: string;
  /**
   * Button type
   */
  type?: 'button' | 'submit' | 'reset';
  /**
   * Button title.
   */
  title: string;

  disabled: boolean;
}