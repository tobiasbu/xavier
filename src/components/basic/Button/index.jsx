import React from 'react';
import PropTypes from 'prop-types';

/**
 * @typedef { import("./types").ButtonProps } ButtonProps
 * @typedef { import("./types").Button } Button
 */

/**
 * Button component
 * @param { ButtonProps } props Button properties
 */
const Button = (props) => {
  const { children, color } = props;
  let text = children;
  if (children) {
    text = children.trim();
    if (text.length <= 0) {
      text = Button.defaultProps.children;
    }
  }

  return (
    <button
      className={`a-btn a-btn--${color} a-btn--medium`}
      type="button"
      aria-label={`${text}`}
    >
      {text}
    </button>
  );
};

Button.defaultProps = {
  // Easter-Egg to remember that we NEED to insert text in a button.
  // By the way, this song is from Led Zeppelin. (DON'T FORGET TO REMOVE THIS)
  children: 'Ramble on and now\'s the time, the time is now, to sing my song...',
  color: 'uranus',
};

Button.propTypes = {
  children: PropTypes.string,
  color: PropTypes.string,
};

export default Button;
