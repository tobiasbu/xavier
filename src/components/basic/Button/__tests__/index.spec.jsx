import React from 'react';
import renderer from 'react-test-renderer';
import Button from '..';

describe('Astro Button', () => {
  it('common render', () => {
    const component = renderer.create(<Button title="Listen to LZ">Ramble on and now s the time, the time is now, to sing my song</Button>);
    const compJson = component.toJSON();
    expect(compJson).toMatchSnapshot();
  });

  it('with color', () => {
    const component = renderer.create(<Button title="Green Button" color="earth">Earth</Button>);
    const compJson = component.toJSON();
    expect(compJson).toMatchSnapshot();
  });
});
