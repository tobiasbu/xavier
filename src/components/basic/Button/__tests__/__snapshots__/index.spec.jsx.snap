// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Astro Button common render 1`] = `
<button
  aria-label="Ramble on and now s the time, the time is now, to sing my song"
  className="a-btn a-btn--uranus a-btn--medium "
  disabled={false}
  title="Listen to LZ"
  type="button"
>
  Ramble on and now s the time, the time is now, to sing my song
</button>
`;

exports[`Astro Button with color 1`] = `
<button
  aria-label="Earth"
  className="a-btn a-btn--earth a-btn--medium "
  disabled={false}
  title="Green Button"
  type="button"
>
  Earth
</button>
`;
